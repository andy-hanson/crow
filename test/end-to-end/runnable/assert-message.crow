main void() unsafe
	(() => assert 1::nat + 1 == 3) log-error
	(() => forbid 1::nat + 1 == 2) log-error
	(() => assert false : "Custom assert message") log-error
	(() => forbid true : "Custom forbid message") log-error

log-error void(cb act void()) unsafe
	match cb catch
	as ok ()
		error log "Should throw"
	as error e
		error log "{e}"
