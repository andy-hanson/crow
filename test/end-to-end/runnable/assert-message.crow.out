exit code: 0
stderr:
error: Asserted expression is false: 1::nat + 1 == 3
	at main
	at call
	at catch
	at call
	at catch
	at catch
	at log
	at main
error: Forbidden expression is true: 1::nat + 1 == 2
	at main
	at call
	at catch
	at call
	at catch
	at catch
	at log
	at main
error: Custom assert message
	at main
	at call
	at catch
	at call
	at catch
	at catch
	at log
	at main
error: Custom forbid message
	at main
	at call
	at catch
	at call
	at catch
	at catch
	at log
	at main
